lib/signer.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+
+/**
+ * @typedef {signer}
+ * @function sign
+ */
+
+
+/**
+ * Creates signer.
+ *
+ * @param {AuthorizationBuilder} AuthorizationBuilder
+ * @return {signer}
+ */
+module.exports = (AuthorizationBuilder) => {
+    /**
+     * @param {string} code
+     * @param {string} secret
+     * @param {requestOptions} requestOptions
+     * @param {string|stream~Readable} body
+     *
+     * @return {Promise}
+     */
+    function sign(code, secret, requestOptions, body) {
```

Comment:
I am not sure how this would work with our current ContentHasher. Will have to look into this. Going to have to add a ticket for this as well. This will not be a trivial task. This streaming stuff was the bulk of the time so far so I can't imagine this approach will be much easier.